### PR TITLE
Prepare for running the dist suite as an unprivileged user

### PR DESCRIPTION
**Pull request by Xanewok**
Uncontroversial bits of #128

Let's land this first to minimize bitrot and let #128 focus strictly on sandboxing details (need to update it to use fuse-overlayfs and separate another binary entrypoint for the unshared userns variant of the sandbox)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -88,9 +86,94 @@ pub fn main() {
...
+    }
+
+    let start_coordinator = env::var("CACHEPOT_START_COORDINATOR").is_ok()
+        || env::args_os().any(|a| a == "--start-coordinator");
+
+    // TODO: That's a rough heuristic - share detection logic from cmdline.rs
+    let kind = match (std::env::args().nth(1).as_deref(), start_coordinator) {
+        (Some("scheduler"), _) => Kind::DistScheduler,
+        (Some("worker"), _) => Kind::DistWorker,
+        (_, true) => Kind::Coordinator,
```
Comment (drahnr) [inline src/lib.rs:136]: Not: Splitting the coordinator from the triple in an if case makes this easier to read imho

File: systemd/config/scheduler.conf [hunk 2]
```diff
@@ -1,6 +1,6 @@
 public_addr = "http://127.0.0.1:10600"
 
-[server_auth]
+[worker_auth]
```
Comment (drahnr) [inline systemd/config/scheduler.conf:3]: ,👍

**Review by drahnr — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (Xanewok) [reply src/lib.rs]: Fixed in 938054adfb540ce3942a6793ab3e146f50ee42bb

**Comment by Xanewok** [comment]
Er, did I fat-finger something? I enabled the auto-merge but I don't remember merging this by hand and the other checks are still running... In any case, that was not intentional! But the CI was green before so there's that :sweat_smile:

**Pull request merged**